controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -0,0 +1,292 @@
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=get
+
+// Reconcile reconciles the StorageAutoscaler object
+func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.Log.Info("reconciling storage autoscaler", "namespace", request.Namespace, "name", request.Name)
+
+	// list the storage autoscaler reconciler object
+	storageAutoScaler := &ocsv1.StorageAutoScaler{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, storageAutoScaler)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("storage autoscaler not found for reconcile request", "namespace", request.Namespace, "name", request.Name)
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		r.Log.Error(err, "failed to get storage autoscaler")
+		return reconcile.Result{}, err
+	}
+
+	// if no expansion is in-progress update the phase to "not-started"
+	if storageAutoScaler.Status.Phase == "" {
+		originalStorageAutoScaler := storageAutoScaler.DeepCopy()
+		storageAutoScaler.Status.Phase = "NotStarted"
+		err := r.updateStatus(ctx, storageAutoScaler, originalStorageAutoScaler)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	// if the expansion is InProgress or Failed, verify the scaling
+	if storageAutoScaler.Status.Phase == "InProgress" || storageAutoScaler.Status.Phase == "Failed" {
+		return r.verifyScaling(ctx, storageAutoScaler)
+	}
+
+	// if the last expansion is succeeded 10 minutes ago, do not scale
+	if storageAutoScaler.Status.LastExpansion != nil {
+		if time.Since(storageAutoScaler.Status.LastExpansion.CompletionTime.Time).Seconds() < 600 {
+			err := fmt.Errorf("last expansion was succeeded less than 10 minutes ago")
+			r.Log.Error(err, "scaling cannot be performed")
+			// return nil error as scraper will retry if needed
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// list the storagecluster
+	storageCluster := &ocsv1.StorageCluster{}
+	storageClusterName := storageAutoScaler.Spec.StorageCluster.Name
+	err = r.Get(ctx, client.ObjectKey{Name: storageClusterName, Namespace: request.Namespace}, storageCluster)
+	if err != nil {
+		r.Log.Error(err, "failed to get storage cluster")
+		return reconcile.Result{}, err
+	}
+
+	// list the storagecluster deviceset storageclass
+	err = r.detectLsoStorageclass(ctx, storageCluster)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// check if the ceph cluster is healthy
+	notHealthy, err := r.checkIfCephClusterIsNotHealthy(ctx, storageCluster)
+	if err != nil {
+		r.Log.Error(err, "failed to check if ceph cluster is healthy")
+		return reconcile.Result{}, err
+	}
+	if notHealthy {
+		r.Log.Info("ceph cluster is not healthy cannot scale")
```

Comment:
This seems like an error, or at least a warning

Reply:
there is no warning in ocs operator, so we can raise a error